src/processors/modulation/Flanger.cpp

```diff
@@ -0,0 +1,300 @@
+    createTimeMsParameter (params, "delayAmount", "Amount", delayAmountRange, 2.0f);
+    createTimeMsParameter (params, "delayOffset", "Offset", delayOffsetRange, 1.0f);
+    createPercentParameter (params, "feedback", "Feedback", 0.0f);
+    createPercentParameter (params, "mix", "Mix", 0.5f);
+
+    emplace_param<AudioParameterChoice> (params, delayTypeTag, "Delay Type", StringArray { "Clean", "Lo-Fi" }, 0);
+
+    return { params.begin(), params.end() };
+}
+
+void Flanger::prepare (double sampleRate, int samplesPerBlock)
+{
+    dsp::ProcessSpec spec { sampleRate, (uint32) samplesPerBlock, 2 };
+    dsp::ProcessSpec monoSpec { sampleRate, (uint32) samplesPerBlock, 1 };
+    fs = (float) sampleRate;
+
+    for (int ch = 0; ch < 2; ++ch)
+    {
+        for (int i = 0; i < delaysPerChannel; ++i)
+        {
+            cleanDelay[ch][i].prepare (monoSpec);
+            lofiDelay[ch][i].prepare (monoSpec);
+
+            LFOs[ch][i].prepare (monoSpec);
+            LFOData[ch][i].resize ((size_t) samplesPerBlock, 0.0f);
+        }
+
+        smooth[ch].reset (sampleRate, 0.05);
+
+        feedbackState[ch] = 0.0f;
+        fbSmooth[ch].reset (sampleRate, 0.01);
+
+        delaySmoothSamples[ch].reset (sampleRate, 0.01);
+        delayOffsetSmoothSamples[ch].reset (sampleRate, 0.01);
+    }
+
+    // set phase offset
+    constexpr float piOver2 = MathConstants<float>::pi / 2.0f;
+    LFOs[1][0].reset (piOver2);
+
+    aaFilter.prepare (spec);
+    aaFilter.setCutoffFrequency (12000.0f);
+
+    dryWetMixer.prepare (spec);
+    dryWetMixer.setMixingRule (dsp::DryWetMixingRule::sin3dB);
+
+    dcBlocker.prepare (spec);
+    dcBlocker.setCutoffFrequency (60.0f);
+
+    audioOutBuffer.setSize (2, samplesPerBlock);
+    modOutBuffer.setSize (1, samplesPerBlock);
+
+    for (auto& filt : hilbertFilter)
+        filt.reset();
+
+    bypassNeedsReset = false;
+}
+
+void Flanger::processModulation (int numSamples)
+{
+    modOutBuffer.setSize (1, numSamples, false, false, true);
+    if (inputsConnected.contains (ModulationInput))
+    {
+        // get modulation buffer from input (-1, 1)
+        const auto& modInputBuffer = getInputBuffer (ModulationInput);
+        BufferHelpers::collapseToMonoBuffer (modInputBuffer, modOutBuffer);
+
+        auto phaseShiftLFO = [this, numSamples, filterIndex = 0] (float* lfoIn, float* lfoOut) mutable
+        {
+            // trying to use more of the Hilbert filters than we have available!
+            jassert (filterIndex < (int) std::size (hilbertFilter));
+
+            auto& filter = hilbertFilter[filterIndex++];
+            for (int n = 0; n < numSamples; ++n)
+                std::tie (lfoIn[n], lfoOut[n]) = filter.process (lfoIn[n]);
+        };
+
+        auto* modOutData = modOutBuffer.getReadPointer (0);
+
+        FloatVectorOperations::copy (LFOData[0][0].data(), modOutData, numSamples);
+        phaseShiftLFO (LFOData[0][0].data(), LFOData[0][1].data());
```

Comment:
```suggestion
        phaseShiftLFO (LFOData[0][0].data(), LFOData[1][0].data());
```
I think that puts everything in range? Also, we no longer need two Hilbert filters, since now we're only using one.

Reply:
Oh right, thanks for catching that!